### ISSUE TITLE
Share Link Requires Login and then reentering the Link

### ISSUE THREAD

**Issue by sstemann** [NCATSTranslator/Feedback#521]
I dont know how it should work, but the Share Link
<img width="447" alt="image" src="https://github.com/NCATSTranslator/Feedback/assets/38321826/f3a08201-b442-4ecd-9c94-ff198939372c">

https://ui.test.transltr.io/main/results?l=Diabetes Mellitus&i=MONDO:0005015&t=0&q=546a71ad-59e0-4a98-b7b2-af9276817ae6 - this space in the middle is a real issue for GitHub, copy and paste "https" to "6" in your browser

it only accessible after you have logged in.  the steps were:

1.  NIH MFA Chrome - run a query 
2. Share the Link [https://ui.test.transltr.io/main/results?l=Diabetes Mellitus&i=MONDO:0005015&t=0&q=546a71ad-59e0-4a98-b7b2-af9276817ae6](https://ui.test.transltr.io/main/results?l=Diabetes Mellitus&i=MONDO:0005015&t=0&q=546a71ad-59e0-4a98-b7b2-af9276817ae6)
3. Open an Incognito Chrome window that has not been logged into Translator and paste Link URL in the browser
4. Go through the process of logging in with Google in the Incognito Window
5. Logged in, but shared link results are not displayed
6. Paste (again) the Link URL in the browser

**Comment by gglusman** [comment]
Many websites encode the intended URL as a redirect from the login. Would be nice to have indeed!

**Comment by dnsmith124** [comment]
@gglusman this is a feature that has been on our list to implement since we first started on user accounts, unfortunately it just didn't make the cut for this release. But it is planned to be implemented soon!

Looking in the code the url is properly encoded, but somewhere in the process of it being output on the page as HTML it appears to render with the spaces rather than %20 as it should. Working on resolving that now.

**Comment by dnsmith124** [comment]
I've pushed a fix that's live on CI that fixes the encoding issue with urls, they should now be properly displaying as encoded both in the box that's visible in the UI and when you click the "copy link" button.

There is a larger change to be made here though, I'll try to detail it as best I can:
Essentially the frontend populates the `/main` or `/demo` portion of the url based on the current url in the address bar. This means that if you accessed one of the example diseases while **not** logged in, the share url would show `/demo`, and thus not require a login if shared with someone else. The issue with this implementation is that even when you access an example disease while logged in, then attempt to share those results, the share url will use `/main`, which will require the user to sign in to access it even if the query itself is an example query.

In short, I need to add in functionality that will detect whether a query's curie is among the examples (the functionality of which is currently handled entirely by the UI backend) and dynamically set the share url to demo whenever this is the case.

**Comment by sstemann** [comment]
resolved!
